infra/aws/terraform/kops-infra-ci/main.tf

```diff
@@ -14,18 +14,67 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// This configures Google GKE cluster OIDC issuer as an Identity
-// Provider (IdP), and allows the list of audiences specified.
-resource "aws_iam_openid_connect_provider" "google_prow_idp" {
-  provider = aws.kops-infra-ci
+terraform {
+  required_version = "~> 1.1"
 
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.11"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "~> 2.20"
+    }
+  }
 
-  url            = "https://container.googleapis.com/v1/projects/k8s-prow/locations/us-central1-f/clusters/prow"
-  client_id_list = ["sts.amazonaws.com"]
+  backend "s3" {
+    bucket = "k8s-infra-kops-ci-tf-state"
+    region = "us-east-2"
+    key    = "kops-infra-ci/terraform.tfstate"
+    # TODO(ameukam): stop used hardcoded account id. Preferably use SSO user
+    role_arn     = "arn:aws:iam::808842816990:role/OrganizationAccountAccessRole"
+    session_name = "kops-infra-ci"
+  }
+}
+
+provider "aws" {
+  region = local.region
+
+  assume_role {
+    role_arn = "arn:aws:iam::${local.kops-infra-ci-account-id}:role/OrganizationAccountAccessRole"
+  }
+}
+
+provider "kubernetes" {
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+
+  # This requires the awscli to be installed locally where Terraform is executed.
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks.cluster_name]
+  }
+}
+
+################################################################################
+# Common Locals
+################################################################################
+
+locals {
+  region = "us-east-2"
 
-  # AWS wants the thumbprint of the the top intermediate certificate authority.
-  thumbprint_list = [
-    # GlobalSign root certificate (Google Managed Certficates)
-    "08745487e891c19e3078c1f2a07e452950ef36f6"
-  ]
+  kops-infra-ci-name       = "kops-infra-ci"
+  kops-infra-ci-index      = index(data.aws_organizations_organization.current.accounts[*].name, local.kops-infra-ci-name)
+  kops-infra-ci-account-id = data.aws_organizations_organization.current.accounts[local.kops-infra-ci-index].id
```

Comment:
this too suffers from the same cyclical loop issue

![image](https://github.com/kubernetes/k8s.io/assets/10913471/3359205f-cdd2-485c-9eb4-14dbdd03d410)